### PR TITLE
Redux component

### PR DESCRIPTION
**Pull request by Pi1998**
# Branch Changes 

- Wrapped `<App>` with `<Provider>`
- Passed `store` into `<Provider>`
- Included `initialState`
- Used `books` in component
- Used `useDispatch` action
- Added event handler to add a book and to remove a book

### CODE REVIEW

**Comment by addod19** [comment]
## Hi @Pi1998, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the linters are not set up correctly

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: book-store/src/components/addBook.js [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+    e.target.title.value = '';
+    e.target.author.value = '';
+  };
+
+  return (
+    <div className="bookInput">
+      <h3 className="addBook">Add new book</h3>
+      <form className="bookForm" onSubmit={submitHandler}>
+        <input type="text" className="title" name="title" placeholder="Book title" />
+        <input type="text" className="author" name="author" placeholder="Author" />
```
Comment (Johnadibe) [inline book-store/src/components/addBook.js]: > - [ ] Hello @Pi1998, I noticed that you did not implement the `onChange` function for each form input in the `AddBook` component. While this is a good practice, Adding an `onChange` event handler allows users to see and edit the input field content in real-time. Without it, users won't have visibility into what they are typing. Also, By using `onChange`, you can create controlled components in React. Controlled components provide a more predictable and reliable way to manage form input values. I suggest you add a local state for `title` and `author` which is updated through the `onChange` event in the AddBook component to improve user interaction and data handling.👍😊

File: book-store/src/components/addBook.js [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+        <button type="submit" className="btnAdd">Add</button>
```
Comment (Johnadibe) [inline book-store/src/components/addBook.js]: > - [ ] Based on the project requirements, please create a custom <Button> component with an event handler that adds a book to the books array when clicked. You can use this custom <Button> component in your AddBook component for better adherence to the project specifications.👍
> 
> ad
> 
> You can use the below as a guide:
> ```
> const Button = ({ onClick, children }) => {
>   return (
>     <button onClick={onClick} className="your-button-styles">
>       {children}
>     </button>
>   );
> };
> 
> ```
> 
> ```
> import Button from './Button'; // Import your custom Button component
> 
> function AddBook() {
> // Your dispatch and selector variable
> 
>   // Other logic that controls form state
> 
>   return (
>     <div>
>       {/* Rest of form JSX */}
>       <Button onClick={submitHandler}>
>         Add book
>       </Button>
>     </div>
>   );
> }
> ```

File: book-store/src/components/book.js [hunk 3]
```diff
@@ -1,35 +1,41 @@
...
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <div className="description">
+        <p className="category">{bookInfo.category}</p>
+        <h3 className="title">{bookInfo.title}</h3>
+        <small className="author">{bookInfo.author}</small>
+        <div className="btnContainer">
+          <button type="button" className="remove" onClick={() => dispatch(removeBook(bookInfo.item_id))}>Remove</button>
```
Comment (Johnadibe) [inline book-store/src/components/book.js]: > - [ ] Based on the project requirements, please create a custom <Button> component with an event handler that removes a book from the books array when clicked. You can use this custom <Button> component in your Book component for better adherence to the project specifications.👍
> 
> ol
> 
> You can use the below as a guide:
> ```
> const Button = ({ onClick, children }) => {
>   return (
>     <button onClick={onClick} className="your-button-styles">
>       {children}
>     </button>
>   );
> };
> 
> ```
> 
> ```
> import Button from './Button'; // Import your custom Button component
> 
> function Book() {
> // Your dispatch variable
> 
> const handleRemove = () => {
>   dispatch(removeBook(bookInfo.item_id));	
> };
> 
>   return (
>     <div>
>       {/* Rest of your JSX code */}
> <div className="btnContainer">
>        <Button onClick={handleRemove}>
>         Remove
>       </Button>
>          <button type="button" className="edit">
>            Edit
>         </button>
>        </div>
>      {/* Rest of your JSX code */}
>     </div>
>   );
> }
> ```

File: book-store/src/redux/books/booksSlice.js [hunk 4]
```diff
@@ -1,18 +1,37 @@
...
+  ],
 };
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
   reducers: {
     addBook: (state, action) => {
-      state.books.push(action.payload);
+      state.booksAry.push(action.payload);
```
Comment (Johnadibe) [inline book-store/src/redux/books/booksSlice.js]: > - [OPTIONAL] I noticed in your code that you're directly mutating the state by pushing the action.payload into the state.booksAry. While this approach might work, it's generally not considered a good practice to directly mutate the state in Redux. Instead, I recommend using the Redux Toolkit's createSlice to handle state updates in an immutable way. You can modify your addBook reducer like this:
> 
> ```
> addBook: (state, action) => {
>   state.booksAry = [...state.booksAry, action.payload];
> },
> ```
>

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @Pi1998👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Linter checks passed✔️
✅ Wrapped the <App> component with the <Provider> component from react-redux✔
✅ Passed the configured store into the <Provider>✔
✅ Create an initialState variable for booksSlice✔
✅ Displayed books received from the slice, in a reusable component✔
✅ Dispatched actions using useDispatch✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @Pi1998   👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
✅ Made suggested changes 👍 
✅ used Redux store to display books 👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**